Jbeorse/refactor ui

jbeorse opened the pull request:
Clean up and refactor front end code. Add templating.

jbeorse committed:
Standardize object formatting

jbeorse committed:
Merge branch 'master' of github.com:mtkp/lemongrab

jbeorse committed:
Moved strict mode to function level

jbeorse committed:
Move event handler declarations to initialization

jbeorse committed:
Refactor public API format

jbeorse committed:
Added Handlebars lib

jbeorse committed:
Add Handlebars templating. Reduce to a single list

jbeorse committed:
Merge branch 'master' of github.com:mtkp/lemongrab

jbeorse committed:
Fixed trailing white space

jbeorse committed:
Fix variable declarations

mtkp commented on lemongrab/static/js/todo.js:
```diff
@@ -1,132 +1,118 @@
   var _createTask = function(id, description) {
-    var parent = $(constants.incompleteList),
-        task, control;
+    var parent  =   $("#task-list"),
```
any particular reason for the spacing here?

jbeorse replied:
The style guide I mentioned had a tab before and after the "=" for all variables. We're going to be declaring every variable with a single var statement, so we're trying to mak it visuall easy to scan the full list of them. The tabs are to align the variable names and instantiations vertically. Of course, I don't have any methods that have multiple variables instantiated at the top. 

Style of course is a preference and we don't need to use that. Or we could only apply it when there are multiple variable instantiations. Would you prefer that?

mtkp commented on lemongrab/static/js/todo.js:
```diff
@@ -1,132 +1,118 @@
-    }).appendTo(control);
+  // Mark task as completed
+  var _completeTask = function(id) {
+    var task  = $("#task-" + id);
```
same here re spacing

mtkp commented on lemongrab/static/js/todo.js:
```diff
@@ -1,132 +1,118 @@
-        task = $("#" + constants.taskIdPrefix + id);
+  // Redo a task
+  var _redoTask = function(id) {
+    var task  = $("#task-" + id);
```
same here re spacing

mtkp commented on lemongrab/templates/index.html:
```diff
@@ -22,20 +25,28 @@
+        <div class="task-body">{{description}}</div>
+        <div class="task-incomplete">Not completed</div>
+        <div class="task-control">
+          <input type="button" id="toggle" value="Complete"
```
Not sure how much it matters but you could end these inputs with a slash like on line 29,  `<input />`

jbeorse replied:
Good catch.

In HTML5 the closing '/' is depracated and shouldn't be used for void elemets. It's supported in all browsers and won't cause a problem, but its really an ignored syntax error. So this instance is correct and the others are wrong. Here's google's style guide's statement on the matter: http://google-styleguide.googlecode.com/svn/trunk/htmlcssguide.xml?showone=Document_Type#Document_Type

I'll fix them.

mtkp replied:
Wow... and here I thought we were supposed to be closing them. Good to know!

jbeorse committed:
Removed closing '/' on void elements

jbeorse committed:
Update var declaration syntax

jbeorse commented:
I updated the var declaration syntax as mentioned in your comments. I guess that made github hide them all. Anyway, I changed it to be the case that "=" are aligned past the end of the longest variable name. Let me know if you don't like this style.

mtkp commented:
Looks good Jeff. I was just curious as to why it was formatted as such. I couldn't see the pattern. But I don't have a problem either way.

jbeorse commented:
No its good, I definitely want to hear it if something is unclear. I'm hoping the code is more readable even to someone without a deeper understanding of what's going on.
